src/app/components/PersistGate.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { PersistGate } from 'redux-persist/es/integration/react';
+
+import { history, persistor, store } from '../store/configureStore';
+import { reloginAction, selectToken } from '../../common/services/user';
+
+const EnhancedPersistGate = ({ children }) => (
```

Comment:
We are using the persist gate but this component is actually something else. It has some meaning and function in the app beyond the gatepersistor. It takes care of relogin. I am not sure about the right name but i know persistgate isnt right. Do you have any idea?

Reply:
That would be if we had the persist gate solely for the relogin action.
If thats the case, we can drop it completely and move the relogin action to saga which will fire on router location changed.

If we need the persist gate, then this name is totally ok, as we are doing partial application on the original persist gate... just packed it into a module.

Using relogin-specific name would hide the fact in the Root.js that we have persist gate installed...

Reply:
we need the persist for the relogin only i think. although its good to have because in real world app this will be a place to do other app-related stuff we need to do before redux is hydrated

Reply:
if you insist on keeping the name we should at least call it `StorePersistGate`